service/group0_state_broadcaster.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * Modified by ScyllaDB
+ * Copyright 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#include "group0_state_broadcaster.hh"
+#include "compaction/compaction_manager.hh"
+#include "replica/database.hh"
+
+namespace service {
+
+namespace {
+
+logging::logger slogger("group0_state_broadcaster");
+
+} // namespace
+
+void group0_state_broadcaster::start() {
+    SCYLLA_ASSERT(this_shard_id() == 0);
+
+    if (_stopped) {
+        return;
+    }
+
+    _done = make_ready_future<>();
+
+    _timer.set_callback([this] {
+        slogger.debug("Disseminating last state id ...");
+
+        _done = sys_ks()
+                    .get_last_group0_state_id()
+                    .then([this](const utils::UUID& state_id) {
+                        return gossiper().add_local_application_state(gms::application_state::STATE_ID,
+                                                                      gms::versioned_value::state_id(service::state_id(state_id)));
+                    })
+                    .then([this] {
+                        utils::UUID min_state_id;
+                        gossiper().for_each_endpoint_state([this, &min_state_id](const gms::inet_address& addr, const gms::endpoint_state& eps) {
+                            if (gossiper().is_left(addr)) {
+                                // ignoring the left nodes
+                                return;
+                            }
+
+                            const utils::UUID state_id = std::invoke([state_id_value = eps.get_application_state_ptr(gms::application_state::STATE_ID)] {
```

Comment:
I am curios why use `std::invoke` as opposite of calling the lambda directly?

Reply:
reworked